File: isPalindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if (s.length() <= 1) return false;
        transform(s.begin(), s.end(), s.begin(), ::tolower);
        auto left = s.begin();
        auto right = prev(s.end());
        while (left != right) {
             if (!::isalnum(*left)) left++;
             else if(!::isalnum(*right)) right--;
             else if(*left != *right) return false;
        }

        return true;
    }
};

File: isPalindrome.java
public class Solution {
    public boolean isPalindrome(int x) {
        if ( x < 0) return false;
        int d = 1;

        // should be >= 10, since 1, 2, ..., 9
        while (x / d >= 10) {
            d *= 10;
        }

        while (x > 0) {
            int q = x / d;
            int r = x % 10;

            if (q != r) return false;

            x = x % d;
            x /= 10;
            d /= 100;
        }

        return true;
    }
}